pkg/containerd/opts/spec.go

```diff
@@ -42,6 +42,12 @@ func WithRelativeRoot(root string) oci.SpecOpts {
 	}
 }
 
+// WithoutRoot sets the root to nil for the container.
+func WithoutRoot(ctx context.Context, client oci.Client, c *containers.Container, s *runtimespec.Spec) error {
```

Comment:
Don't you need this for an Argon? Does the shim handle this case?

Reply:
Your change always set it. :) https://github.com/jterry75/cri/blob/windows_port/pkg/server/container_create_windows.go#L282

So for Argon, I guess hcsshim assumes rootfs is under `cwd/rootfs`? If that is the case, I'll set it here now.

BTW, does hcsshim support readonly rootfs?

Reply:
Not for windows: https://github.com/microsoft/hcsshim/blob/master/internal/hcsoci/hcsdoc_wcow.go#L177

But also I dont see how my change works for argons :). See here: https://github.com/microsoft/hcsshim/blob/master/internal/hcsoci/hcsdoc_wcow.go#L189

Reply:
You must of tested this though so if it works lets leave it. I wonder if containerd is setting the volume GUID automatically

Reply:
Looks like the shim forces it here: https://github.com/microsoft/hcsshim/blob/d3edaf5e605e418ff37b1dd12c13481507506c2d/cmd/containerd-shim-runhcs-v1/service_internal.go#L136

Reply:
> Not for windows: https://github.com/microsoft/hcsshim/blob/master/internal/hcsoci/hcsdoc_wcow.go#L177

I'll add comment about this in the code. :)

Reply:
Done